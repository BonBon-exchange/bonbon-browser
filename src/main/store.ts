/* eslint-disable no-template-curly-in-string */
/* eslint-disable import/prefer-default-export */
import Store from 'electron-store';

const store = new Store();

export const getStore = () => store;

if (store.get('browsing.defaultWebpage') === undefined)
  store.set('browsing.defaultWebpage', 'https://www.google.com');

if (store.get('browsing.searchEngine') === undefined)
  store.set('browsing.searchEngine', 'DuckDuckGo');

if (store.get('browsing.customSearchEngine') === undefined)
  store.set(
    'browsing.customSearchEngine',
    'https://duckduckgo.com/?q=${search}'
  );

if (store.get('browsing.width') === undefined) store.set('browsing.width', 600);

if (store.get('browsing.height') === undefined)
  store.set('browsing.height', 800);

if (
  store.get('browsing.size') === undefined ||
  store.get('browsing.size') === 'fit' // fix related with issue #341 https://github.com/BonBon-exchange/bonbon-web-browser/issues/341
)
  store.set('browsing.size', 'lastResized');

if (store.get('browsing.topEdge') === undefined)
  store.set('browsing.topEdge', 'fit');

// migrate searchEngine
const searchEngineValue = store.get('browsing.searchEngine') as string;
const searchEngineMigrationObject: Record<string, string> = {
  google: 'Google',
  presearch: 'Presearch',
  qwant: 'Qwant',
  duckduckgo: 'DuckDuckGo',
  yandex: 'Yandex',
  swisscows: 'Swisscows',
  ecosia: 'Ecosia',
  startpage: 'Startpage',
};
const newSearchEngineValue =
  searchEngineMigrationObject[searchEngineValue] || searchEngineValue;
store.set('browsing.searchEngine', newSearchEngineValue);
