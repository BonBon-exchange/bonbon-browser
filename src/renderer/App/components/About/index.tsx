import { About } from './About';

export * from './About';
export default About;
