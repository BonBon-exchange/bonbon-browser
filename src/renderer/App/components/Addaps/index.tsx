import { Addaps } from './Addaps';

export * from './Addaps';
export default Addaps;
