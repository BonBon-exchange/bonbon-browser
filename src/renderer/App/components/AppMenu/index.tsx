import { AppMenu } from './AppMenu';

export * from './AppMenu';
export default AppMenu;
