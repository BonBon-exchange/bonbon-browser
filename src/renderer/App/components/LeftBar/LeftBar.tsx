/* eslint-disable jsx-a11y/alt-text */
/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable import/prefer-default-export */
import React from 'react';

import { useBoard } from 'renderer/App/hooks/useBoard';
import { BrowserProps } from 'renderer/App/components/Browser/Types';
import { useBrowserMethods } from 'renderer/App/hooks/useBrowserMethods';

import './style.scss';

export const LeftBar: React.FC = () => {
  const board = useBoard();
  const { focus } = useBrowserMethods();

  return (
    <div className="LeftBar__browserFavContainer">
      {board?.browsers.map((b: BrowserProps) => {
        return (
          <div
            className="LeftBar__browserFav"
            key={b.id}
            onClick={() => focus(document, b.id)}
          >
            <img
              data-tip={b.title}
              src={b.favicon}
              className="LeftBar__browserFavImg"
              data-browserid={b.id}
            />
          </div>
        );
      })}
    </div>
  );
};
