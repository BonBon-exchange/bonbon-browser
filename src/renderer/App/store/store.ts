import { configureStore, Store } from '@reduxjs/toolkit';
import {
  persistStore,
  persistReducer,
  createMigrate,
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
  Persistor,
} from 'redux-persist';
import storage from 'redux-persist/lib/storage';

import boardReducer from './reducers/Board';
import downloadsReducer from './reducers/Downloads';
import { migrations } from './migrations';

export const store: Store = configureStore({
  reducer: {
    board: boardReducer,
    downloads: downloadsReducer,
  },
  devTools: process.env.NODE_ENV !== 'production',
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export const getPersistedStoreAndPersistor = (
  id: string
): { persistor: Persistor; store: Store } => {
  const persistConfig = {
    key: id,
    version: 3,
    storage,
    migrate: createMigrate(migrations, { debug: true }),
  };
  const persistedBoard = persistReducer(persistConfig, boardReducer);
  const persistedDownloads = persistReducer(persistConfig, downloadsReducer);
  const persistedStore = configureStore({
    reducer: {
      board: persistedBoard,
      downloads: persistedDownloads,
    },
    devTools: process.env.NODE_ENV !== 'production',
    middleware: (getDefaultMiddleware) =>
      getDefaultMiddleware({
        serializableCheck: {
          ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
        },
      }),
  });
  const persistor = persistStore(persistedStore);

  return { persistor, store: persistedStore };
};
