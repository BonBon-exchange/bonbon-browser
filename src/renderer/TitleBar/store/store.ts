import { configureStore } from '@reduxjs/toolkit';
import {
  persistStore,
  persistReducer,
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';
import storage from 'redux-persist/lib/storage';

import tabsReducer from './reducers/Tabs';

const persistConfig = {
  key: 'titleBar',
  version: 1,
  storage,
};

const persistedTabs = persistReducer(persistConfig, tabsReducer);

export const store = configureStore({
  reducer: {
    tabs: persistedTabs,
  },
  devTools: process.env.NODE_ENV !== 'production',
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: {
        ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
      },
    }),
});

export const persistor = persistStore(store);

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
