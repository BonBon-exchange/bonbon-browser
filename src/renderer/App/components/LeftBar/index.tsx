import { LeftBar } from './LeftBar';

export * from './LeftBar';
export default LeftBar;
