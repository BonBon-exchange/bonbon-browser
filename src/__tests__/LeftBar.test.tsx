/* eslint-disable react/jsx-props-no-spreading */
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';

import { LeftBar } from '../renderer/App/components/LeftBar';
import { mockWindow } from './beforeAll';
import {
  addBrowser,
  removeBrowser,
} from '../renderer/App/store/reducers/Board';
import { store } from '../renderer/App/store/store';

const addBrowserAction = {
  id: 'randomid',
  url: 'https://www.github.com',
  top: 0,
  left: 0,
  height: 800,
  width: 600,
  firstRendering: true,
  favicon: '',
};

describe('LeftBar', () => {
  beforeAll(() => {
    mockWindow();
  });

  it('should render', () => {
    expect(
      render(
        <Provider store={store}>
          <LeftBar />
        </Provider>
      )
    ).toBeTruthy();
  });

  it('should have one browser because of initial state', () => {
    const { container } = render(
      <Provider store={store}>
        <LeftBar />
      </Provider>
    );

    expect(container.getElementsByClassName('LeftBar__browserFav').length).toBe(
      1
    );
  });

  it('should have two browsers in the store', () => {
    store.dispatch(addBrowser(addBrowserAction));
    expect(store.getState().board.board.browsers.length).toBe(2);
  });

  it('should have two browsers in dom', () => {
    const { container } = render(
      <Provider store={store}>
        <LeftBar />
      </Provider>
    );

    expect(container.getElementsByClassName('LeftBar__browserFav').length).toBe(
      2
    );
  });

  it('should have one browser in the store', () => {
    store.dispatch(removeBrowser(addBrowserAction.id));
    expect(store.getState().board.board.browsers.length).toBe(1);
  });

  it('should have 1 browser in dom', () => {
    const { container } = render(
      <Provider store={store}>
        <LeftBar />
      </Provider>
    );

    expect(container.getElementsByClassName('LeftBar__browserFav').length).toBe(
      1
    );
  });
});
