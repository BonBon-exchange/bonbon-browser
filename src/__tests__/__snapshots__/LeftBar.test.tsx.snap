// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeftBar should render 1`] = `
<div
  id="LeftBar__browserFavContainer"
>
  <ul
    style={Object {}}
  >
    <div
      id="LeftBar__browserFavContainerItems"
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="AddIcon"
      focusable="false"
      id="ButtonAddBrowser"
      onClick={[Function]}
      viewBox="0 0 24 24"
    >
      <path
        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
      />
    </svg>
  </ul>
</div>
`;
